test/ResourceCacheSpec.js

```diff
@@ -135,5 +135,27 @@ describe('ResourceCache', function() {
         }
       });
     });
+
+    describe('when a uuid is provided', function() {
+      it('updates an existing record', function(done) {
+        var cache = getCache();
+        cache.persist({ foo: 'bar' }).then(function(records) {
+          var uuid = records[0].uuid;
+          cache.persist({ uuid: uuid, foo: 'bar2' }).then(function(updatedRecords) {
+            if (updatedRecords.length === 1 &&
+                updatedRecords[0].uuid === uuid &&
+                updatedRecords[0].is_dirty == true &&
+                updatedRecords[0].created_at.valueOf() === records[0].created_at.valueOf() &&
+                updatedRecords[0].updated_at !== records[0].updated_at.valueOf() &&
+                updatedRecords[0].hasOwnProperty('foo') &&
```

Comment:
Initially, I was using `Object.create` to "clone" the record I was passing back, but it doesn't work that way. It just creates a new object and sets its prototype. So you can call `myObject.foo` and it will go up the prototype chain to get the value. I added this check in there to guard against the same mistake. Now it will fail unless the new record has its own `foo` property.